### PR TITLE
Allow user to cherry pick scopes.

### PR DESCRIPTION
**Pull request by Zegnat**
This presents the requested scopes in a list with checkboxes so the user can uncheck any they do not wish to grant. The default is to have all the requested scopes checked.

First step towards #26.

Screenshot: new interface showing checkboxes for scopes.

### CODE REVIEW

**Review by sebsel — COMMENTED** [review]
It might be too much, but in my testing, when I used a different userURL, the form seems to accept it. Of course, the password will not match etc, so there will be no signed code, but maybe we can display the username, or, since we only support one, check for a valid one before showing the form?

Bit unrelated to this PR maybe, I can open a new issue.

**edit**: moving this to it's own issue

**Review by sebsel — DISMISSED** [review]
Tested a bit with all kinds of scopes, looks good to me.

File: index.php
Comment (Zegnat) [inline index.php:277]: I had an unexpected output trying to use an emoji as scope. Suggesting this does not return `false` when one of the scopes is wrong, but will update the `scopes` array to contain `false`. This means the following check against `false` does nothing and the `implode()` will actually add extra spaces.

Needs some more testing before merging!

**Review by Zegnat — COMMENTED** [review]
**DO NOT MERGE**

File: index.php
Comment (Zegnat) [reply index.php:277]: This is addressed by 17bdeea.

**Comment by Zegnat** [comment]
If you have time, @sebsel, please rerun your scope tests to make sure my new commit did not break anything.

**Comment by sebsel** [comment]
Would be nice to have some automation in these tests. I don't know exactly what I did last time 0:)

**Review by sknebel — APPROVED** [review]
reviewed with @Zegnat at IWC Berlin, LGTM

**Pull request merged**
